src/Base/index.js

```diff
@@ -1,78 +1,93 @@
 import React from 'react';
-import {ActivityIndicator, Text, View, Pressable} from 'react-native';
+import {Text, View, TextInput} from 'react-native';
 
-const Button = ({
-  disabled = false,
-  onPress,
-  busy = false,
-  loaderIcon = <ActivityIndicator color="#fff" />,
-  busyText = 'Processing..',
+const Input = ({
+  onChangeText,
+  iconPosition,
+  icon,
+  style,
+  value,
+  placeholder = 'Enter Text',
+  label = null,
+  error,
   ...props
 }) => {
-  const textStyles = {
-    ...(props.textStyles || {
-      fontWeight: 'bold',
+  const [focused, setFocused] = React.useState(false);
+  const textInputStyles = {
+    ...(props.textInputStyles || {
+      flex: 1,
+      width: '100%',
       color: props.textColor || '#fff',
       letterSpacing: 0,
     }),
   };
+  const getFlexDirection = () => {
+    if (icon && iconPosition) {
+      if (iconPosition === 'left') {
+        return 'row';
+      } else if (iconPosition === 'right') {
+        return 'row-reverse';
+      }
+    }
+  };
+
+  const getBorderColor = () => {
+    if (error) {
+      return '#DC2626';
+    }
+
+    if (focused) {
+      return '#E4E4E7';
+    }
+  };
 
   return (
-    <Pressable
-      disabledd={disabled}
-      onPress={() => {
-        if (!disabled) {
-          onPress();
-        }
-      }}
-      style={({pressed}) => [
-        {
-          backgroundColor: pressed
-            ? props.onPressColor || '#27272A'
-            : props.backgroundColor || '#18181B',
-          width: '100%',
-          borderColor: props.borderColor || '#27272A',
-          borderWidth: 1,
-          borderRadius: 4,
-          paddingVertical: 8,
-          paddingHorizontal: 4,
-          alignItems: 'center',
-          justifyContent: 'center',
-        },
-      ]}>
-      {busy === false && (
-        <View>
-          {typeof props.children === 'string' ? (
-            <Text style={textStyles}>{props.children}</Text>
-          ) : (
-            props.children
-          )}
-        </View>
-      )}
+    <View style={{paddingVertical: 12}}>
```

Comment:
Only one thing left - `paddingVertical: 12` 

This would become hardcoded

Reply:
@Dhruv-Rathi

Reply:
`<View style={{...(props.labelStyles || {paddingVertical: 12})}}>`
@rjvim  This is fine?

Reply:
@Dhruv-Rathi I think there are not label styles but the styles for entire Wrapper. I think we should put any styles at all for this. 

What I mean is, first of all, padding or margin what's needed would be the issue of the screen which will use our `<TextInput />` component. So, someone who wants to use our component, should wrap it in something like:

```
<View style={{ marginVertical: 12 }}>
<TextInput />
</View>
```


Is it possible to entirely remove "View" wrapping?